sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/PartitioningAwareFileIndex.scala

```diff
@@ -221,6 +221,15 @@ abstract class PartitioningAwareFileIndex(
         if (!fs.isDirectory(userDefinedBasePath)) {
           throw new IllegalArgumentException(s"Option '$BASE_PATH_PARAM' must be a directory")
         }
+        def qualifiedPath(path: Path): Path = path.makeQualified(fs.getUri, fs.getWorkingDirectory)
+
+        val qualifiedBasePath = qualifiedPath(userDefinedBasePath)
+        rootPaths
+          .find(p => !qualifiedPath(p).toString.startsWith(qualifiedBasePath.toString))
+          .foreach { rp =>
+            throw new IllegalArgumentException(
+              s"Wrong basePath $userDefinedBasePath for the root path: $rp")
+          }
         Set(fs.makeQualified(userDefinedBasePath))
```

Comment:
Can be simply `Set(qualifiedBasePath)` as we now calculated it before; if we want to change `qualifiedPath()` to return String, `Set(new Path(qualifiedBasePath))`.

Reply:
Good idea!